pkgs/stdenv/linux/default.nix

```diff
@@ -347,6 +350,14 @@ in
         };
       };
 
+      # force gmp to rebuild so we have the option of dynamically linking
+      # libgmp without creating a reference path from:
+      #   stage5.gcc -> stage4.coreutils -> stage3.glibc -> bootstrap
+      gmp = lib.makeOverridable (super.gmp.override { stdenv = self.stdenv; }).overrideAttrs (a: { pname = "${a.pname}-stage4"; });
+
+      # coreutils gets rebuilt both here and also in the final stage; we rename this one to avoid confusion
```

Comment:
I think it's not true that it gets rebuilt in the final stage.

Reply:
You are correct.

I'm unable to reproduce the situation I was experiencing two months ago, where coreutils was getting built twice.  Unfortunately I don't have much more time today, so I think the best course of action is to simply revert the commit that added the `"-stage4"` marker.

I'll have more time to look at this tomorrow morning.

I'm starting to suspect that the problem may have been caused by my own site-specific overlays, which have changed a lot in the last two months.

Reply:
I figured out the reason.  We should still revert the commit (i.e. we should merge #179961).

The two-builds-of-coreutils phenomenon is caused by #168983, and only on powerpc64le.  I will explain in more detail tomorrow.  Sorry for not realizing this earlier.